lib/utils.js

```diff
@@ -767,6 +776,36 @@ exports.stackTraceFilter = function() {
   };
 };
 
+//
+/**
+ * Number.MAX_SAFE_INTEGER only available in ES2015.
+ * @see https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
+ * @type {number}
+ */
+exports.MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || 0x1fffffffffffff;
+
+/**
+ * Attempt to convert `value` an integer, if it's a representation
+ * of a "safe" unsigned integer.
+ * @param {*} [value] Value to convert
+ * @see https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/parseInt
+ * @returns {(number|NaN)} result
+ */
+exports.toSafeUnsignedInteger = function toSafeUnsignedInteger(value) {
+  var intVal;
+  if (/^0[xX]/.test(value)) {
+    intVal = parseInt(value, 16);
+  } else if (/^0[^.]/.test(value)) {
+    intVal = parseInt(value, 8);
+  } else {
+    intVal = parseInt(value, 10);
+  }
+  return !isNaN(intVal) && parseFloat(intVal) === intVal
+    && exports.isSafeUnsignedInteger(intVal)
+    ? intVal
+    : NaN;
```

Comment:
`isSafeUnsignedInteger` already checks for an integer so there's no need to double check here. Also if the value is `NaN`, you'll still hit the right checks without the additional `NaN` check.

```suggestion
  return exports.isSafeUnsignedInteger(intVal) ? intVal : NaN;
```